src/Editor.js

```diff
@@ -482,43 +484,130 @@ const Editor = () => {
           Undo
         </button>
         <button
+          draggable
           onClick={() => {
             setPointer(pointer + 1);
           }}
         >
           Redo
         </button>
         <PanelTitle>Layers</PanelTitle>
-        <ol>
-          {doc.layers.map(({ id, name }) => (
-            <li
-              key={id}
-              style={{
-                color: selection.has(id) ? "#f0f" : null,
-                cursor: "pointer",
-                padding: "6px",
-                borderBottom: "1px solid #ddd"
-              }}
-              onClick={event => {
-                event.stopPropagation();
-                setState(
-                  current => ({
-                    ...current,
-                    selection:
-                      keys.has("ShiftLeft") || keys.has("ShiftRight")
-                        ? current.selection.has(id)
-                          ? not(current.selection, [id])
-                          : or(current.selection, [id])
-                        : set([id])
-                  }),
-                  true
-                );
-              }}
-            >
-              {name}
-            </li>
-          ))}
-        </ol>
+        <DragDropContext
+          onDragStart={result => {
+            const { draggableId } = result;
+            setElementBeingDraggedId(draggableId);
+            setState(current => {
+              return {
+                ...current,
+                selection:
+                  current.selection.size === 0
+                    ? set([draggableId])
+                    : current.selection
+              };
+            });
+          }}
+          onDragEnd={result => {
+            const { destination, source, draggableId } = result;
+
+            setElementBeingDraggedId("");
+            // destination may be null if you drag outside of the droppable area.
+            if (
+              !destination ||
+              (destination.droppableId === source.droppableId &&
+                destination.index === source.index)
+            ) {
+              return;
+            }
+
+            const layers = [...doc.layers];
+            const draggedLayer = layers.splice(source.index, 1)[0];
+            layers.splice(destination.index, 0, draggedLayer);
```

Comment:
We've already got a handy little function for this!

https://github.com/haydn/sketchbook-js/blob/master/src/reorder.js

Reply:
Might need to be modified to handle reordering multiple items at once? Probably worthwhile putting some unit tests around it to cover it current behaviour and then take a TDD approach to make it support multiple layers at a time.